readthedocs/search/faceted_search.py

```diff
@@ -96,53 +95,112 @@ class PageSearchBase(RTDFacetedSearch):
     }
     doc_types = [PageDocument]
     index = PageDocument._doc_type.index
-    fields = ['title^10', 'headers^5', 'content']
+
+    outer_fields = ['title^10']
+    section_fields = ['sections.title^5', 'sections.content']
+    domain_fields = [
+        'domains.type_display',
+        'domains.doc_display',
+        'domains.name',
+        'domains.display_name',
+    ]
+    fields = outer_fields
+
+    # need to search for both 'and' and 'or' operations
+    # the score of and should be higher as it satisfies both or and and
     operators = ['and', 'or']
 
+    def query(self, search, query):
+        """Manipulates query to support nested query."""
+        search = search.highlight_options(encoder='html', number_of_fragments=3)
 
-class DomainSearchBase(RTDFacetedSearch):
-    facets = {
-        'project': TermsFacet(field='project'),
-        'version': TermsFacet(field='version'),
-        'role_name': TermsFacet(field='role_name'),
-    }
-    doc_types = [SphinxDomainDocument]
-    index = SphinxDomainDocument._doc_type.index
-    fields = ('display_name^5', 'name^3', 'project^3', 'type_display')
-    operators = ['and']
+        # match query for the title (of the page) field.
+        match_title_query = Match(title=query)
+
+        # nested query for search in sections
+        sections_nested_query = self.generate_nested_query(
+            query=query,
+            path='sections',
+            fields=self.section_fields,
+            inner_hits={
```

Comment:
Can we not use the DSL `highlight` function here, because it's a nested query?

Reply:
Yes.
Here's a example of it -- https://github.com/elastic/elasticsearch-dsl-py/issues/960